### ISSUE TITLE
[Async] Add async types to Player base types for asset level compatibility

### ISSUE THREAD

**Issue by sugarmanz** [player-ui/player#467]
The Player content validator works by consuming Player & Player plugin types. In order to support async assets (using `AsyncNodePlugin`), we'll need to export the async spec (below) to the base Player types:

```json
{
  "id": "some-async-asset",
  "async": true
}
```

**Comment by KetanReddy** [comment]
The base Player types right now don't export a lot of specific information about the inner workings of Player so if we didn't want to define the Async Node type there, we could add it as a transform to in the LSP to just add the async node info to the `AssetWrapperOrSwtich` type when loaded

**Comment by sugarmanz** [comment]
That would be interesting -- given that this is functionality powered by an optional plugin, I'd almost opt for this strategy so that we don't conflate core Player types with superfluous types? Is there a way we could configure the validator to conditionally handle async assets if enabled?

**Comment by KetanReddy** [comment]
Yep! The LSP supports adding transforms for XLR types that will be loaded in so we could distribute a plugin to do that. It would just be required to load that plugin before loading in any types but that is already a documented requirement.
